src/classes/LD_LeadConvertOverride_CTRL.cls

```diff
@@ -421,11 +445,29 @@ public with sharing class LD_LeadConvertOverride_CTRL {
             // if we created a new company account, affiliate to it.
             if (strAccountId == 'NEW')
                 strAfflAccountId = lcr.getAccountId();
-                
+            
+            // if we created a new company acct and the Lead had a company address, set the new acct's address
+            if (strAccountId == 'NEW' && !String.isBlank(l.CompanyStreet__c)) {
+            	update new Account(
+                    Id=lcr.getAccountId(),
+                    BillingStreet = l.CompanyStreet__c, 
+                    BillingCity = l.CompanyCity__c,
+                    BillingState = l.CompanyState__c,
+                    BillingPostalCode = l.CompanyPostalCode__c,
+                    BillingCountry = l.CompanyCountry__c
+                );
+            }   
             // handle the optional Affiliation to a new or existing Account.
             ID afflAccId = null;
             if (strAfflAccountId == 'NEW') {
-                Account acc = new Account(name=l.Company);
+                Account acc = new Account(
```

Comment:
did we do any testing in orgs with State & Country codes enabled?  I know we have some helper routines that deal with setting state & country that first tries the code, and then sets the text if the code doesn't match.  If you don't use those routines, then if State & Country codes are enabled, but the user put in "Washington" for the State, the update will fail since it has to be "WA" instead.